### PR TITLE
docs: add migration guide for theme data

### PR DESCRIPTION
**Pull request by ArsalaBangash**
@Mister-Hope recommended adding docs about themeConfig access to the v2 migration guide after he answered [this question](https://github.com/vuepress/vuepress-next/discussions/686)

### CODE REVIEW

File: docs/guide/migration.md [hunk 1]
```diff
@@ -291,6 +291,7 @@ Some major breaking changes:
   - Files in `theme/layouts/` directory will not be registered as layout components automatically. You need to specify it explicitly in `layouts` option.
   - Files in `theme/templates/` directory will not be used as dev / ssr template automatically.
   - Always provide a theme entry file, and do not use `"main": "layouts/Layout.vue"` as the theme entry.
+- To access the themeConfig as you would via `this.$site.themeConfig` in v1, we now recommend using the [@vuepress/plugin-theme-data](../reference/plugin/theme-data.md) plugin and it's useThemeData compoition function
 - Stylus is no longer the default CSS pre-processor, and the stylus palette system is not embedded. If you still want to use similar palette system as v1, [@vuepress/plugin-palette](../reference/plugin/palette.md) may help.
-- Markdown code blocks syntax highlighting by Prism.js is not embedded by default. You can use either [@vuepress/plugin-prismjs](../reference/plugin/prismjs.md) or [@vuepress/plugin-shiki](../reference/plugin/shiki.md), or implement syntax highlighting in your own way.
```
Comment (Mister-Hope) [inline docs/guide/migration.md:295]: I think `embedded` is better, Meteorlxy is trying to express we are extracting code higlight feature as plugins from `@vuepress/markdown` (the markdown render) here.

File: docs/guide/migration.md [hunk 2]
```diff
@@ -291,6 +291,7 @@ Some major breaking changes:
   - Files in `theme/layouts/` directory will not be registered as layout components automatically. You need to specify it explicitly in `layouts` option.
   - Files in `theme/templates/` directory will not be used as dev / ssr template automatically.
   - Always provide a theme entry file, and do not use `"main": "layouts/Layout.vue"` as the theme entry.
+- To access the themeConfig as you would via `this.$site.themeConfig` in v1, we now recommend using the [@vuepress/plugin-theme-data](../reference/plugin/theme-data.md) plugin and it's useThemeData compoition function
```
Comment (Mister-Hope) [inline docs/guide/migration.md:294]: Please add the below line to zh docs.

如果你想要在主题中访问 themeConfig，我们现在建议使用 [@vuepress/plugin-theme-data](../reference/plugin/theme-data.md) 插件和它提供的 Composition API `useThemeData`

Comment (Mister-Hope) [inline docs/guide/migration.md:294]:
```suggestion
- To access the themeConfig as you would via `this.$site.themeConfig` in v1, we now recommend using the [@vuepress/plugin-theme-data](../reference/plugin/theme-data.md) plugin and it's `useThemeData` compoition function
```

**Review by Mister-Hope — CHANGES_REQUESTED** [review]

**Review by meteorlxy — CHANGES_REQUESTED** [review]

**Comment by meteorlxy** [comment]
@ArsalaBangash Thanks for the contribution! ❤️

**Pull request merged**